### PR TITLE
Add clock into intrinsic api into wasm runtime

### PR DESCRIPTION
**Pull request by bianpengyuan**
Add a new API method `getCurrentTimeNanoseconds` utilizing enovy's time source.

### CODE REVIEW

**Comment by bianpengyuan** [comment]
@jplevyak @PiotrSikora ptal thank you!

File: source/extensions/common/wasm/wasm.h [hunk 1]
```diff
@@ -54,6 +54,7 @@ using WasmCallback3Void = void (*)(void*, uint32_t, uint32_t, uint32_t);
 using WasmCallback4Void = void (*)(void*, uint32_t, uint32_t, uint32_t, uint32_t);
 using WasmCallback5Void = void (*)(void*, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t);
 using WasmCallback0Int = uint32_t (*)(void*);
+using WasmCallback0Int64 = uint64_t (*)(void*);
```
Comment (jplevyak) [inline source/extensions/common/wasm/wasm.h]: For more complex signatures we are moving to the standard c++ name mangling.  See https://github.com/jplevyak/envoy/blob/3f8ae2c418104ea7132f89a04dd1f0db3058fd2a/source/extensions/common/wasm/wasm.h#L68

File: source/extensions/common/wasm/wasm.cc [hunk 2]
```diff
@@ -1126,6 +1136,10 @@ void Wasm::tickHandler() {
   }
 }
 
+uint64_t Wasm::getCurrentTimeNanoseconds() {
+  return std::chrono::duration_cast<std::chrono::nanoseconds>(time_source_.systemTime().time_since_epoch()).count();
```
[the remark this answers is not in the document]
Comment (bianpengyuan) [reply source/extensions/common/wasm/wasm.cc]: Done.

File: source/extensions/common/wasm/wasm.h (hunk 1 above)
Comment (bianpengyuan) [reply source/extensions/common/wasm/wasm.h]: Is WasmCallback_mj the correct mangling for uint64_t (void*)?

Comment (jplevyak) [reply source/extensions/common/wasm/wasm.h]: yes.  I added comments to the stats-meta PR: see https://itanium-cxx-abi.github.io/cxx-abi/abi.html#mangling-builtin, but TLDR: Z = void, j = uint32_t, l = int64_t, m = uint64_t  
where the return type comes first.

**Comment by bianpengyuan** [comment]
@jplevyak Rebased. Would you mind to take another look and help to merge this? thanks!

**Pull request merged**